hail/src/main/scala/is/hail/expr/types/physical/PCanonicalStruct.scala

```diff
@@ -37,6 +37,39 @@ object PCanonicalStruct {
 
   def canonical(t: Type): PStruct = PType.canonical(t).asInstanceOf[PStruct]
   def canonical(t: PType): PStruct = PType.canonical(t).asInstanceOf[PStruct]
+
+  def getByteSizeAndOffsets(types: Array[PType], nMissingBytes: Long, byteOffsets: Array[Long]): Long = {
```

Comment:
don't move this. In a separate PR, we should rename PBaseStruct to PCanonicalBaseStruct, then introduce an abstract PBaseStruct above PStruct and PTuple as we discussed.

Reply:
https://github.com/hail-is/hail/issues